src/index.ts

```diff
@@ -0,0 +1,119 @@
+import net from "net";
+import {BeaconState} from "./types";
+import * as msgs from "./messages";
+import Peer, {PeerOpts} from "./peer";
+
+export interface HobbitsOpts {
+  port?: number,
+  networkId: number,
+  chainId: number,
+  bootnodes?: string[],
+  state: BeaconState
+}
+
+export default class HobbitsP2PNetwork {
+  private port: number;
+  private networkId: number;
+  private chainId: number;
+  private peers: Peer[];
+  private bootnodes: string[];
+  private state: BeaconState;
+  private server: net.Server;
+
+  /**
+   * Constructor
+   * @param {HobbitsOpts} opts
+   */
+  public constructor(opts: HobbitsOpts) {
+    this.port = opts.port || 9000;
+    this.networkId = opts.networkId;
+    this.chainId = opts.chainId;
+    this.state = opts.state;
+    this.peers = [];
+    this.bootnodes = opts.bootnodes || [];
+    this.server = net.createServer();
+  }
+
+  /**
+   * Connects to static peers
+   */
+  private connectStaticPeers = () => {
+    this.bootnodes.map((bootnode: string) => {
+      this.connect(bootnode);
+    });
+    console.log(`Connected to ${this.peers.length} static peers`);
+  };
+
+  /**
+   * Connects to a peer
+   * @param {string} ip
+   */
+  private connect = (ip: string) => {
+    const peer: Peer = new Peer({
+      ip,
+      port: this.port
+    });
+
+    const res: boolean = peer.connect();
```

Comment:
This should await connection, once connect is made async. In current scenario, connect would always return true even in case of failed connection